### PR TITLE
SDK - Compiler - Made compiler compatible with @wraps

### PR DESCRIPTION
**Pull request by Ark-kun**
Fixes https://github.com/kubeflow/pipelines/issues/3367

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -793,7 +792,8 @@ def _create_workflow(self,
       raise ValueError('Either specify pipeline params in the pipeline function, or in "params_list", but not both.')
 
     args_list = []
-    for arg_name in argspec.args:
+    signature = inspect.signature(pipeline_func)
```
Comment (Bobgy) [inline sdk/python/kfp/compiler/compiler.py:795]: nit: adding a test case? this seems like a very subtle change

Comment (Ark-kun) [reply sdk/python/kfp/compiler/compiler.py:795]: I'm not sure we need to explicitly support `@wraps`. It's just that the old functions like `getfullargspec` do not understand it. Those functions were planned to be deprecated, but then were kept.

Comment (Bobgy) [reply sdk/python/kfp/compiler/compiler.py:795]: got it, SGTM

**Comment by Bobgy** [comment]
FYI, cherry picking to 1.0 branch

**Pull request merged**